Orange/data/table/base.py

```diff
@@ -0,0 +1,1383 @@
+            # explicitly construct a sparse table as this can be called from Table(...),
+            # but allow subclasses to call correctly
+            if issubclass(cls, SparseTable):
+                return cls._from_sparse_numpy(domain, X, Y, metas, weights)
+            else:
+                return SparseTable._from_sparse_numpy(domain, X, Y, metas, weights)
+
+        if domain is None:
+            result = cls._from_data_inferred(X, Y, metas)
+            if weights is not None:
+                result.set_weights(weights)
+            return result
+
+        # ensure correct shapes (but not sizes) so we can iterate
+        def correct_shape(what):
+            if what is None or what.ndim == 2:
+                return what
+            else:
+                return np.atleast_2d(what).T
+
+        X = correct_shape(X)
+        Y = correct_shape(Y)
+        metas = correct_shape(metas)
+
+        result = cls()
+        for role_array, variables in ((X, domain.attributes), (Y, domain.class_vars), (metas, domain.metas)):
+            if role_array is None:
+                if variables:
+                    raise ValueError("Variable and column count mismatch. ")
+                continue
+            if role_array.shape[1] != len(variables):
+                raise ValueError("Variable and column count mismatch. ")
+            for column, variable in zip(role_array.T, variables):
+                result[variable.name] = column
+        result.domain = domain
+        if weights is not None:
+            result.set_weights(weights)
+
+        result._transform_discrete_values_to_descriptors()
+        result._transform_discrete_into_categorical()
+        result._transform_timevariable_into_datetime()
+
+        return result
+
+    @classmethod
+    def from_list(cls, domain, rows, weights=None):
+        """Construct a table from a list of rows.
+
+        Parameters
+        ----------
+        domain : Domain
+            The domain to use for the data.
+        rows : list[list[obj]]
+            A list of rows. Must be rectangular with the number of columns
+            matching the columns in the domain.
+        weights : anything TableBase.set_weights accepts, optional, default None
+            The weights to use for the resulting Table.
+
+        Returns
+        -------
+        TableBase
+            A Table constructed from the given data.
+        """
+        if weights is not None and len(rows) != len(weights):
+            raise ValueError("Mismatching number of instances and weights.")
+        # check dimensions, pandas raises a very nondescript error
+        row_width = len(rows[0])
+        for r in rows:
+            if len(r) != row_width:
+                raise ValueError("Inconsistent number of columns.")
+
+        # check row lengths
+        domain_columns = len(domain.variables) + len(domain.metas)
+        for r in rows:
+            if len(r) != domain_columns:
```

Comment:
Can you replace these for loops with `any()` call and also batch the two checks together so it's a single iteration.